tools/device.c

```diff
@@ -137,43 +143,114 @@ static int bl_device__is_ttyACM(const struct dirent *entry)
 	return strncmp(entry->d_name, "ttyACM", 6) == 0;
 }
 
-static int bl_device__scan(void)
+static bool bl_device__list_append(
+		struct bl_device **devices,
+		unsigned *count,
+		const char *device_path,
+		const char *device_serial)
+{
+	struct bl_device *temp;
+
+	temp = realloc(*devices, sizeof(**devices) * (*count + 1));
```

Comment:
Probably a nitpick, but I feel like "temp" is a bad variable name. Typically I'd use "ndevices" or something like that.

Reply:
`ndevices` sounds like it would mean the same as `count`!